scale/job/test/test_models.py

```diff
@@ -297,32 +297,47 @@ def test_get_latest_job_exes_with_a_filter(self):
         self.assertDictEqual(latest_job_exes, expected_result, 'latest job executions do not match expected results')
 
     def test_schedule_job_executions(self):
```

Comment:
This test case has become pretty complex. I think it's good to have 1 that mixes different uses together, but it would be nice to have some very simple ones too that are easier to follow. Cases like using no volumes or just using a RO for inputs and no outputs with a single workspace and job type.